go/mysql/conn_test.go

```diff
@@ -38,29 +38,30 @@ func createSocketPair(t *testing.T) (net.Listener, *Conn, *Conn) {
 	wg := sync.WaitGroup{}
 
 	var clientConn net.Conn
+	var clientErr error
 	wg.Add(1)
 	go func() {
 		defer wg.Done()
-		var err error
-		clientConn, err = net.Dial("tcp", addr)
-		if err != nil {
-			t.Fatalf("Dial failed: %v", err)
-		}
+		clientConn, clientErr = net.Dial("tcp", addr)
 	}()
 
 	var serverConn net.Conn
+	var serverErr error
 	wg.Add(1)
 	go func() {
 		defer wg.Done()
-		var err error
-		serverConn, err = listener.Accept()
-		if err != nil {
-			t.Fatalf("Accept failed: %v", err)
-		}
+		serverConn, serverErr = listener.Accept()
 	}()
 
 	wg.Wait()
```

Comment:
Seems like this could block forever if `Dial` fails. Using `DialTimeout` and something like `listener.(*net.TCPListener).SetDeadline(time.Now().Add(...))` to set timeouts might help.

Reply:
Done